oat-sa/extension-tao-mediamanager
# feat/AUT-76/stylesheet-manager-for-passages

**Related to:** [AUT-76](https://oat-sa.atlassian.net/browse/AUT-76)

**Description:**
- Add a Stylesheet manager for passages and isolate the scope of the CSS rules to the passage.
- Attaching/removing/disabling stylesheets to a passage is possible, the same way we can currently in the item authoring tool
- The styles defined with the style editor overwrite the styles defined in the stylesheets (technically, the inline styles from the style editors are applied after)

**Changes:**
The main idea is to keep the CSS file without any modification and add the main passage class to the CSS rules using the CSSDom Browser API. This lets us fake the CSS rules with a link to the file. So if we hide or delete the file, the fake will do the same. So we do load the CSS file and after apply a format function to add this class as a parent to the CSS rules with the CSSDom Browser API.

- Refactor code to add stylesheet manager on Assets panel.
- Changes on Upload files and preview Assets.

**PR related:**

- [ ] https://github.com/oat-sa/extension-tao-itemqti/pull/2143
- [ ] https://github.com/oat-sa/tao-core/pull/3583

**How to check:**

- Check upload files.
- Check download, delete and hide files.
- Check import Asset with CSS files included.
- Check passages with stylesheets on Items and Test.
- Remove file rename option
- Block possibility to add the same file (by name) twice

[Kitchen environment on the ticket.](https://oat-sa.atlassian.net/browse/AUT-76?focusedCommentId=190406)

Disclamer
This feature is using the Feature Service, so it is mandatory to add on `config\tao\client_lib_config_registry.conf.php`
```
        'services/features' => array(
            'visibility' => array(
                'taoMediaManager/creator/StyleSheetManager' => 'show'
            )
        )
